networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -118,17 +126,23 @@ def _allocate_vxlan_segment(self, network, host_id, level, host_config):
             alloc = session.query(AllocationsModel).filter_by(host=host_id, level=level, segment_type=segment_type,
                                                               network_id=network_id).first()
             if alloc:
-                # check if segment is existing, if not the allocation should be deleted and a new one allocated
-                segment = (session.query(ml2_models.NetworkSegment).filter_by(id=alloc.segment_id).first())
+                # check if segment exists, if not the allocation should be deleted and a new one allocated
+                if alloc.segment_id is not None:
+                    segment = (session.query(ml2_models.NetworkSegment).filter_by(id=alloc.segment_id).first())
```

Comment:
(what are the parenthesis for?:)

Reply:
Since seba just indented the line, we'd probably have to ask Andrew about it. There are multiple of those throughout the code, that serve no real purpose, iirc. I'd vote for cleaning them all up at once.

Reply:
Well, if I already touch the line I can also remove them. :)